packages/aws-cdk/lib/api/resource-import/importer.ts

```diff
@@ -1,20 +1,71 @@
-import { DeployOptions } from '@aws-cdk/cloud-assembly-schema';
+import { format } from 'util';
 import * as cfnDiff from '@aws-cdk/cloudformation-diff';
 import { ResourceDifference } from '@aws-cdk/cloudformation-diff';
 import * as cxapi from '@aws-cdk/cx-api';
 import * as chalk from 'chalk';
 import * as fs from 'fs-extra';
 import * as promptly from 'promptly';
-import { assertIsSuccessfulDeployStackResult, Deployments, DeploymentMethod, ResourceIdentifierProperties, ResourcesToImport } from './api/deployments';
-import { Tag } from './api/tags';
-import { StackActivityProgress } from './api/util/cloudformation/stack-activity-monitor';
-import { error, info, success, warning } from './logging';
-import { ToolkitError } from './toolkit/error';
-
-export interface ImportDeploymentOptions extends DeployOptions {
-  deploymentMethod?: DeploymentMethod;
-  progress?: StackActivityProgress;
-  tags?: Tag[];
+import { error, info, warn } from '../../cli/messages';
+import { IIoHost, ToolkitAction } from '../../toolkit/cli-io-host';
+import { ToolkitError } from '../../toolkit/error';
+import { assertIsSuccessfulDeployStackResult, Deployments, DeploymentMethod, ResourceIdentifierProperties, ResourcesToImport } from '../deployments';
+import { Tag } from '../tags';
+import { StackActivityProgress } from '../util/cloudformation/stack-activity-monitor';
+
+export interface ResourceImporterProps {
+  deployments: Deployments;
+  ioHost: IIoHost;
+  action: ToolkitAction;
+}
+
+export interface ImportDeploymentOptions {
+  /**
+   * Name of the toolkit stack to use/deploy
+   *
+   * @default CDKToolkit
+   */
+  readonly toolkitStackName: string;
```

Comment:
is this where you're pointing to when you say "makes toolkitStackName a mandatory option"? outside of 'mandatory option' feeling like an oxymoron, i'm curious why that decision was made and what it means to have a mandatory option with a default.

when i first read 'mandatory option' i thought you meant that you were mandating `toolkitStackName` as an option for cli option bags.

Reply:
It's mandatory on the Toolkit.

Reply:
>  outside of 'mandatory option' feeling like an oxymoron

I guess it's more of a forced choice. You do have options, but you are mandated to pick one. I'm open to learn about better language choices though!